streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StreamTableJoinNode.java

```diff
@@ -29,37 +28,33 @@
 
     private final String[] storeNames;
     private final ProcessorParameters<K, V> processorParameters;
-    private boolean isGlobalKTableJoin;
+    private final String otherJoinSideNodeName;
 
     public StreamTableJoinNode(final String nodeName,
-                        final ProcessorParameters<K, V> processorParameters,
-                        final String[] storeNames) {
+                               final ProcessorParameters<K, V> processorParameters,
+                               final String[] storeNames,
+                               final String otherJoinSideNodeName) {
         super(nodeName,
               false);
 
         // in the case of Stream-Table join the state stores associated with the KTable
         this.storeNames = storeNames;
         this.processorParameters = processorParameters;
+        this.otherJoinSideNodeName = otherJoinSideNodeName;
     }
 
-    String[] storeNames() {
-        return Arrays.copyOf(storeNames, storeNames.length);
-    }
-
-    ProcessorParameters<K, V> processorParameters() {
-        return processorParameters;
-    }
+    @Override
+    public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
+        final String processorName = processorParameters.processorName();
+        final ProcessorSupplier processorSupplier = processorParameters.processorSupplier();
 
-    public void setGlobalKTableJoin(boolean globalKTableJoin) {
-        isGlobalKTableJoin = globalKTableJoin;
-    }
+        // Stream - GlobalKTable join
+        topologyBuilder.addProcessor(processorName, processorSupplier, parentNode().nodeName());
```

Comment:
This is done for both join, right? (ie, update comment?)

Reply:
ack